### PR TITLE
Package dependency updates

### PR DESCRIPTION
**Pull request by kbwestfall**
Deprecates support for python <= 3.10.  I also updated other package dependencies, trying to at least synchronize them to when 3.11 was released.

This also includes some doc fixes.

We'll need to run tests.

### CODE REVIEW

**Comment by kbwestfall** [comment]
PS - This also includes some updates to the `installing` doc.

File: tox.ini [hunk 1]
```diff
@@ -1,12 +1,12 @@
 [tox]
 envlist =
-    {3.10,3.11,3.12}-test{,-alldeps,-shapely,-specutils}{,-cov}
-    {3.10,3.11,3.12}-test-numpy{123,124,125,126}
-    {3.10,3.11,3.12}-test-{numpy,astropy,linetools,ginga}dev
+    {3.11,3.12}-test{,-alldeps,-shapely,-specutils}{,-cov}
+    {3.11,3.12}-test-numpy{124,125,126,200}
```
Comment (tbowers7) [inline tox.ini]: NumPy 2.1 has been released... do we want to test against it, too?

File: tox.ini [hunk 2]
```diff
@@ -36,19 +36,19 @@ description =
     devdeps: with the latest developer version of key dependencies
     oldestdeps: with the oldest supported version of key dependencies
     cov: and test coverage
-    numpy123: with numpy 1.23.*
     numpy124: with numpy 1.24.*
     numpy125: with numpy 1.25.*
     numpy126: with numpy 1.26.*
+    numpy200: with numpy 2.0.*
```
Comment (tbowers7) [inline tox.ini:42]: NumPy 2.1 has been released... do we want to test against it, too?

File: tox.ini [hunk 3]
```diff
@@ -36,19 +36,19 @@ description =
...
 
 # The following provides some specific pinnings for key packages
 deps =
 
     cov: coverage
-    numpy123: numpy==1.23.*
     numpy124: numpy==1.24.*
     numpy125: numpy==1.25.*
     numpy126: numpy==1.26.*
+    numpy200: numpy==2.0.*
```
Comment (tbowers7) [inline tox.ini:52]: NumPy 2.1 has been released... do we want to test against it, too?

File: tox.ini (hunk 1 above)
Comment (kbwestfall) [reply tox.ini]: Fair point.  Added!

**Review by debora-pe — APPROVED** [review]
Thanks @kbwestfall!!

**Review by badpandabear — APPROVED** [review]

**Review by rcooke-ast — APPROVED** [review]
Do you also need to change `tox.ini`? Otherwise, this all looks great to me - thanks for doing this!

**Comment by kbwestfall** [comment]
> Do you also need to change `tox.ini`? Otherwise, this all looks great to me - thanks for doing this!

Can you say more?  The PR has changes to `tox.ini`, but did I miss something?

**Comment by rcooke-ast** [comment]
> > Do you also need to change `tox.ini`? Otherwise, this all looks great to me - thanks for doing this!
> 
> Can you say more? The PR has changes to `tox.ini`, but did I miss something?

Ahhh... sorry - I didn't see it at the bottom of the PR. All looks good to me!!

**Comment by kbwestfall** [comment]
> > > Do you also need to change `tox.ini`? Otherwise, this all looks great to me - thanks for doing this!
> > 
> > 
> > Can you say more? The PR has changes to `tox.ini`, but did I miss something?
> 
> Ahhh... sorry - I didn't see it at the bottom of the PR. All looks good to me!!

Phew!  Thanks for checking.

**Pull request merged**